test/trento_web/controllers/v1/discovery_controller_test.exs

```diff
@@ -32,5 +40,27 @@ defmodule TrentoWeb.V1.DiscoveryControllerTest do
       })
       |> json_response(202)
     end
+
+    test "collect action discards application instance registrations when the associated database does not exists",
+         %{conn: conn} do
+      body =
+        load_discovery_event_fixture("sap_system_discovery_application")
+
+      expect(Trento.Commanded.Mock, :dispatch, fn _ ->
+        {:error, :any_error}
+      end)
+
+      %{status: status} =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post("/api/v1/collect", body)
+
+      assert status != 202
```

Comment:
**question:** the endpoint returns 500. Is it intended?

Reply:
TL;DR We shouldn't care

The response code is inherited by the error returned by the dispatcher, which the fallback controller then handles. AS `:any_error` is not handled, `500` is returned.

Reply:
Anyway, thanks to your comment I realized that the test might be weak, because the controller may fail for other reasons that the declared in the test definition.

I narrowed the case in https://github.com/trento-project/web/pull/3040/commits/d55d837204c1dc8199c0783a8e83c4cb85751f91

Reply:
Correct, thanks. In reality it returns the 404 :+1: